pkg/virt-launcher/virtwrap/agent_poller.go

```diff
@@ -0,0 +1,188 @@
+package virtwrap
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
+)
+
+// TODO: make this configurable?
+const startDelay = 120
+
+type AgentPoller struct {
+	Connection cli.Connection
+	VmiUID     types.UID
+	domainData DomainData
+	active     bool
+	trigger    chan *api.Domain
+	pollTime   int64
+	callback   QemuAgentCallback
+}
+
+type DomainData struct {
+	name       string
+	aliases    []string
+	aliasToMac map[string]string
+	interfaces []api.InterfaceStatus
+}
+
+type Result struct {
+	Page       int
+	Interfaces []Interface `json:"return"`
+}
+type Interface struct {
+	MAC string `json:"hardware-address"`
+	IPs []IP   `json:"ip-addresses"`
+	// TODO: do we want to report the interface name?
+	Name string `json:"name"`
+}
+
+type IP struct {
+	IP     string `json:"ip-address"`
+	Type   string `json:"ip-address-type"`
+	Prefix int    `json:"prefix"`
+}
+
+type QemuAgentCallback func(string, *[]api.InterfaceStatus)
+
+func CreatePoller(connecton cli.Connection, vmiUID types.UID, pollTime int64, callback QemuAgentCallback) *AgentPoller {
+	p := &AgentPoller{
+		Connection: connecton,
+		VmiUID:     vmiUID,
+		pollTime:   pollTime,
+		callback:   callback,
+	}
+	p.create()
+	return p
+
+}
+
+func (p *AgentPoller) create() {
+	p.trigger = make(chan *api.Domain, 5)
+
+	go func() {
+		time.Sleep(time.Duration(startDelay) * time.Second)
+		for {
+			log.Log.Info("Qemu agent poller started")
+			select {
+			case domain := <-p.trigger:
+				if !p.active {
+					continue
+				}
+
+				if domain != nil {
+					p.domainData = p.createDomainData(domain)
+				}
+
+				cmdResult, err := p.pollQemuAgent(p.domainData.name)
+				if err != nil {
+					log.Log.Reason(err).Error("Qemu agent poller error")
+					// TODO: should we fail after a number of retries? (how would we re-enable later?)
+					// should we just increate the break?
```

Comment:
it will be failing in many situations (manually stop service, guest side reboot, etc)

Reply:
Done

Reply:
Leaving as is.